src/query/v2/plan/operator.cpp

```diff
@@ -1311,6 +1360,36 @@ class AggregateCursor : public Cursor {
     }
   }
 
+  void ProcessAll(MultiFrame &multi_frame, ExecutionContext *context) {
+    input_cursor_->PullMultiple(multi_frame, *context);
+    auto valid_frames_modifier =
+        multi_frame.GetValidFramesConsumer();  // consumer is needed i.o. reader because of the evaluator
+    if (valid_frames_modifier.begin() == valid_frames_modifier.end()) {
+      // There are no valid frames, we stop
+      pulled_all_input_ = true;
+      return;
+    }
+
+    for (auto &frame : valid_frames_modifier) {
+      ExpressionEvaluator evaluator(&frame, context->symbol_table, context->evaluation_context, context->request_router,
+                                    storage::v3::View::NEW);
+      ProcessOne(frame, &evaluator);
+    }
+
+    // calculate AVG aggregations (so far they have only been summed)
+    for (size_t pos = 0; pos < self_.aggregations_.size(); ++pos) {
+      if (self_.aggregations_[pos].op != Aggregation::Op::AVG) continue;
+      for (auto &kv : aggregation_) {
+        AggregationValue &agg_value = kv.second;
+        auto count = agg_value.counts_[pos];
+        auto *pull_memory = context->evaluation_context.memory;
+        if (count > 0) {
+          agg_value.values_[pos] = agg_value.values_[pos] / TypedValue(static_cast<double>(count), pull_memory);
```

Comment:
If there are no elements we should still be able to calculate avg, but this seems as temp code, so feel free to do what you think make sense

Reply:
After discussing it on slack we agreed that setting the aggregation value to null is fine and the default value is null (check `EnsureInitialized`), so this is fine.